File: src/clr/src/pal/tests/palsuite/threading/DisableThreadLibraryCalls/test1/testlib.c
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

/*=====================================================================
**
** Source:  testlib.c
**
** Purpose: Simple library that counts thread attach/detach notifications
**
**
**===================================================================*/

#include <palsuite.h>

static int Count;

BOOL __stdcall DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
{

    if (fdwReason == DLL_PROCESS_ATTACH)
    {
        Count = 0;
    }
    else if (fdwReason == DLL_THREAD_ATTACH ||
        fdwReason == DLL_THREAD_DETACH)
    {
        Count++;
    }

    return TRUE;
}

#ifdef WIN32
BOOL __stdcall _DllMainCRTStartup(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
{
    return DllMain(hinstDLL, fdwReason, lpvReserved);
}
#endif

#ifdef WIN32
__declspec(dllexport)
#endif
int __stdcall GetCallCount()
{
    return Count;
}

File: src/tests/JIT/Methodical/cctor/xassem/testlib.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

using System;
namespace Precise
{
    public class measure
    {
        public static int a = 0xCC;
    }
    public class test
    {
        public static byte b1 = 0xC;
        public byte b2 = 0xC;
        public static void f()
        {
        }
        public static void f(ref byte b)
        {
            return;
        }

        static test()
        {
            if (measure.a != 0xCC)
            {
                Console.WriteLine("in .cctor(), measure.a is {0}", measure.a);
                Console.WriteLine("FAILED");
                throw new Exception();
            }
            Console.WriteLine("in .cctor(), measure.a is {0}", measure.a);
            measure.a = 8;
            if (measure.a != 8)
            {
                Console.WriteLine("in .cctor() after measure.a=8, measure.a is {0}", measure.a);
                Console.WriteLine("FAILED");
                throw new Exception();
            }
            Console.WriteLine("in .cctor() after measure.a=8, measure.a is {0}", measure.a);
        }
    }

}

